okio/src/main/java/okio/Okio.java

```diff
@@ -242,4 +242,8 @@ static boolean isAndroidGetsocknameError(AssertionError e) {
     return e.getCause() != null && e.getMessage() != null
         && e.getMessage().contains("getsockname failed");
   }
+
+  public static Source tee(final Source source, final BufferedSink cacheBody) {
```

Comment:
This should be named something other than `cacheBody`. Maybe `copySink`?

Also does it need to be a `BufferedSink`? I think ideally this is just a normal `Sink` (which we may buffer internally).